### PR TITLE
[Date picker][REF] Convert jcalendar date fields to date picker on me…

### PR DESCRIPTION
**Pull request by seamuslee001**
…mber search screen

Overview
----------------------------------------
Switches the member date fields from using jcalendar to datepicker and this tidies up the search template a bit as well

Before
----------------------------------------
Jcalender fields used

Screen Shot 2019-09-03 at 9 25 50 AM

After
----------------------------------------
Datepicker used

Screen Shot 2019-09-03 at 9 25 09 AM

ping @eileenmcnaughton

### CODE REVIEW

**Comment by colemanw** [comment]
Looks like test fails might be related.

**Comment by eileenmcnaughton** [comment]
Also with this we need to ensure _relative gets to 

https://github.com/civicrm/civicrm-core/blob/714ab07093ca650d860ee9e68950c235f7fd78a4/CRM/Contact/BAO/Query.php#L1815
& is processed there - NOT converted into high/low values which will not be loaded correctly from the db when loading saved groups

Old way 
- convert relative dates to high & low & save a separate 'relative_dates' array in formValues in saved search

New way
- retain x_date_relative = 'this.month' when saving. 'relative_dates' array may still be saved (for now) but is irrelevant.

**Comment by seamuslee001** [comment]
ok @eileenmcnaughton should hopefully have fixed that

**Comment by eileenmcnaughton** [comment]
@seamuslee001 the fail relates to the xml change - since this will be stale once we merge the pledge one it might be worth resolving the xml change for this as a separate PR & then rebase this one pledge & that are merged

File: CRM/Member/BAO/Query.php [hunk 1]
```diff
@@ -165,29 +165,36 @@ public static function whereClauseSingle(&$values, &$query) {
     switch ($name) {
       case 'member_join_date_low':
       case 'member_join_date_high':
+      case 'membership_join_date_low':
```
Comment (eileenmcnaughton) [inline CRM/Member/BAO/Query.php]: @seamuslee001 I'm not keen on this - it leaves the deprecated code 'looking useful' - I'd rather separate out the 2 & have deprecation notices in the obsolete code.

We could handle the 'good fields' like this https://github.com/civicrm/civicrm-core/pull/15191

**Comment by seamuslee001** [comment]
@eileenmcnaughton have updated now can you re-check pls

**Comment by eileenmcnaughton** [comment]
OK I think the way the tpl changes close up the  whitespace makes sense  - will add  a  sig-markup label for themers

**Comment by eileenmcnaughton** [comment]
My searches are  working with this but when I went back to check on my previously created & upgraded group I'm seeing  warnings - so the upgrade didn't quite work

Screen Shot 2019-09-03 at 9 31 51 AM

**Comment by seamuslee001** [comment]
@eileenmcnaughton and the upgrade definitely happened?

**Comment by eileenmcnaughton** [comment]
@seamuslee001 yes - but look here https://github.com/civicrm/civicrm-core/blob/6e7cc0f578e814459b1e1b80c7a922684da46140/CRM/Upgrade/Incremental/SmartGroups.php#L168 - we aren't handling _high, _low, _relative in there like I  assumed we would be....

**Comment by seamuslee001** [comment]
@eileenmcnaughton i have just pushed an update to fix the upgrade

**Comment by eileenmcnaughton** [comment]
@seamuslee001 OK -  I was expecting us to make the function handle them but that is fine

**Comment by seamuslee001** [comment]
@eileenmcnaughton we have already used this function before in previous upgrades so i'm more nervous about touching it rather than touching the upgrade code

**Comment by eileenmcnaughton** [comment]
OK - let's merge this but I  have a worry that the legacy group I created didn't save the relative dates - similar to the pledge one - and that feels suspicious to me - like there could have been a previous regression affecting creating new smart groups with relative dates. This 'fixes' but could it be the case that a smart group  created in 5.15 with criteria 'End date' = 'this.month' is NOT working as a relative group - there might be some issue there to track

**Comment by seamuslee001** [comment]
@eileenmcnaughton i have also just pushed up a commit to fix the batch form and some report filters

**Comment by eileenmcnaughton** [comment]
Ah - looks like I messed up the regex here https://github.com/civicrm/civicrm-core/commit/7f175707d67597717d72661c55878d6432f87736#diff-54576153cb6a72d6ea7e22a5e8ca63f2R442

**Pull request merged**